src/state/canisters.rs

```diff
@@ -92,6 +98,28 @@ impl Canisters<true> {
             .clone()
             .expect("Authenticated canisters must have profile details")
     }
+
+    pub fn user_principal(&self) -> Principal {
+        self.identity()
+            .sender()
+            .expect("expect principal to be present")
+    }
+
+    pub async fn is_user_authorized_to_seed_content(&self) -> Result<bool, Box<dyn Error>> {
+        self.content_seed_client
+            .check_if_authorized(self.user_principal())
+            .await
+    }
+
+    pub async fn upload_using_content_seed(
+        &self,
+        delegated_identity: DelegatedIdentityWire,
+        url: String,
+    ) -> Result<(), Box<dyn Error>> {
+        self.content_seed_client
+            .upload_content(url, delegated_identity)
+            .await
+    }
```

Comment:
these functions are unrelated to `Canisters`, content_seed_client should not be in the `Canisters` utility

Reply:
Done